js/models/conversations.js

```diff
@@ -25,6 +25,22 @@
         'blue_grey',
     ];
 
+    function equalArrayBuffers(ab1, ab2) {
+        if (!(ab1 instanceof ArrayBuffer && ab2 instanceof ArrayBuffer)) {
+            return false;
+        }
+        if (ab1.byteLength !== ab2.byteLength) {
+            return false;
+        }
+        var result = true;
+        var ta1 = new Uint8Array(ab1);
+        var ta2 = new Uint8Array(ab2);
+        for (var i = 0; i < ab1.byteLength; ++i) {
```

Comment:
Looks like this is intended to be a constant time comparison.

Consider naming it to `constantTimeEqualArrayBuffers` and maybe adding a comment so it doesn't get "optimized" away by future me.